### PR TITLE
[getdns/flint/arb] Update to the latest version

### LINKED ISSUE #19923: [arb] update to 2.20.0
Library name: arb

New version number: 2.20.0

Other information that may be useful (release notes, etc...)
https://github.com/fredrik-johansson/arb/releases

### LINKED ISSUE #19949: [flint] update to 2.8.0
Library name: flint

New version number: 2.8.0

Other information that may be useful (release notes, etc...)
http://www.flintlib.org/
http://www.flintlib.org/downloads.html

### LINKED ISSUE #19950: [getdns] update to 1.7.0
Library name: getdns

New version number: 1.7.0

Other information that may be useful (release notes, etc...)
https://getdnsapi.net/

### PR DESCRIPTION
**Pull request by NancyLi1013**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #19949 
  Fixes #19950
  Fixes #19923

Update `getdns `to 1.7.0.

Update `flint `to 2.8.0, also update to use cmake to build.

Update `arb` to 2.21.1

Note: All features have passed with the following triplets:

- x86-windows
- x64-windows-static

### CODE REVIEW

**Comment by NancyLi1013** [comment]
Using cmake to build flint, it failed with the error like this:

```
     4>F:\20210803\vcpkg\installed\x64-uwp\include\gmp.h(2240,16): error C4146: unary minus operator applied to unsigned type, result still unsigned (compiling source file F:\20210803\vcpkg\buildtrees\flint\src\flint-2-2f08c3fd05\printf.c) [F:\20210803\vcpkg\buildtrees\flint\x64-uwp-dbg\flint.vcxproj]
     4>F:\20210803\vcpkg\installed\x64-uwp\include\gmp.h(2240,16): error C4146: unary minus operator applied to unsigned type, result still unsigned (compiling source file 
     F:\20210803\vcpkg\buildtrees\flint\src\flint-2-2f08c3fd05\fprintf.c) 
```
So add unsupported uwp for `flint`.

**Comment by NancyLi1013** [comment]
Since `getdns `requires `libunbound` on Linux and osx. Could you please help confirm if we need to add `libunbound-dev` on our CI pipeline? @BillyONeal 

```
CMake Error at /mnt/vcpkg-ci/downloads/tools/cmake-3.21.1-linux/cmake-3.21.1-linux-x86_64/share/cmake-3.21/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Libunbound (missing: LIBUNBOUND_LIBRARIES
  LIBUNBOUND_INCLUDE_DIR) (Required is at least version "1.5.9")
Call Stack (most recent call first):
  /mnt/vcpkg-ci/downloads/tools/cmake-3.21.1-linux/cmake-3.21.1-linux-x86_64/share/cmake-3.21/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  cmake/modules/FindLibunbound.cmake:85 (find_package_handle_standard_args)
  /agent/_work/1/s/scripts/buildsystems/vcpkg.cmake:784 (_find_package)
  CMakeLists.txt:492 (find_package)
```
The source codes are here:

```
# If we're not stub only, we need libunbound.
if (NOT ENABLE_STUB_ONLY)
  find_package(Libunbound "1.5.9" REQUIRED)
  set(HAVE_LIBUNBOUND 1)
  list(APPEND getdns_system_libs Libunbound::Libunbound)

  set(CMAKE_REQUIRED_INCLUDES ${LIBUNBOUND_INCLUDE_DIR})
  set(CMAKE_REQUIRED_LIBRARIES ${LIBUNBOUND_LIBRARIES})

  check_symbol_exists(ub_ctx_set_stub "unbound.h" HAVE_UB_CTX_SET_STUB)

  if (ENABLE_UNBOUND_EVENT_API)
    check_include_file(unbound-event.h HAVE_UNBOUND_EVENT_H)
    check_symbol_exists(ub_ctx_create_ub_event "unbound-event.h" HAVE_UNBOUND_EVENT_API)
  endif ()
else ()
  # Ensure we're not using libunbound items.
  set(HAVE_LIBUNBOUND 0)
  set(HAVE_UNBOUND_EVENT_H 0)
  set(HAVE_UNBOUND_EVENT_API 0)
  set(HAVE_UB_CTX_SET_STUB 0)
endif ()
```

File: ports/flint/portfile.cmake [hunk 1]
```diff
@@ -1,109 +1,33 @@
...
-		RENAME flint.lib
-	)
-	file(INSTALL
-		${SOURCE_PATH}/build.vc${MSVC_VERSION}/dll_flint/${BUILD_ARCH}/Debug/dll_flint.lib
-		DESTINATION ${CURRENT_PACKAGES_DIR}/debug/lib
-		RENAME flint.lib
-	)
-	vcpkg_copy_pdbs()
-else()
-	file(GLOB FLINT_HEADERS "${SOURCE_PATH}/lib/${BUILD_ARCH}/Release/*.h")
```
Comment (JackBoosY) [inline ports/flint/portfile.cmake:86]: Can you please double check the header files whether they need to be changed the dllexport macro?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/flint/portfile.cmake (hunk 1 above)
Comment (NancyLi1013) [reply ports/flint/portfile.cmake:86]: Thanks for pointing this out. Has been checked and updated.

File: ports/arb/portfile.cmake [hunk 2]
```diff
@@ -3,24 +3,23 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
...
-    REF ae6009e3e19bd309a2433467d1b2ddb7001cd1eb # 2.18.1
-    SHA512 78e149f0d51ef8ab29afbad99fd24e3b59acfc509f626e89bdcd57d4a8478b84c3aa51e92f5e26f8a10a20c66d72d2eed50f0dfbfda4a5f5277988f9bac3fa48
+    REF bd68cc0219c5b4bf4a3466b5b6c190203d7dc7b0 # 2.21.0
+    SHA512 5e09ece4855906206c0f5aff9f56d13e1b2432863921af235682b9c45b35e03b1a7db308315bd329e9ed93be36d5e5aac914b1276b9412a551f5f2a189884404
     HEAD_REF master
 )
 
-file(REMOVE ${SOURCE_PATH}/CMakeLists.txt)
-file(COPY ${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt DESTINATION ${SOURCE_PATH})
+file(REMOVE "${SOURCE_PATH}/CMakeLists.txt")
```
Comment (JackBoosY) [inline ports/arb/portfile.cmake:11]: We prefer to use the upstream provided cmake.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Pull request merged**